### PR TITLE
ipatests: Test for ipa hbac rule duplication

### PR DESCRIPTION
**Pull request by menonsudhir**
This testcase checks that default hbac rules
        are not duplicated on the local server when
        ipa-migrate command is run.

### CODE REVIEW

**Comment by rcritten** [comment]
replica[0] isn't installed as an IPA master if  test_remote_server isn't run first.

File: ipatests/test_integration/test_ipa_ipa_migration.py [hunk 1]
```diff
@@ -920,3 +921,28 @@ def test_ipa_migrate_stage_mode_with_invalid_cert(self):
...
+        """
+        run_migrate(
+            self.replicas[0],
+            "prod-mode",
+            self.master.hostname,
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n']
+        )
+        cmd = self.replicas[0].run_command(
```
Comment (rcritten) [inline ipatests/test_integration/test_ipa_ipa_migration.py:939]: A nit I suppose but you aren't constructing a command to run here, you are actually running it. I'd switch 'cmd' with 'result' for clarity.

**Comment by rcritten** [comment]
It is failing because it relies on unexecuted behavior in previous tests. replicas[0] is not a separate IPA server.

File: ipatests/prci_definitions/temp_commit.yaml [hunk 2]
```diff
@@ -72,7 +72,7 @@ jobs:
       class: RunPytest
       args:
         build_url: '{fedora-latest/build_url}'
-        test_suite: test_integration/test_REPLACEME.py
+        test_suite: test_integration/test_ipa_ipa_migration.py::TestIPAMigrateScenario1::test_ipa_hbac_rule_duplication
```
[the remark this answers is not in the document]
Comment (menonsudhir) [reply ipatests/prci_definitions/temp_commit.yaml]: @flo-renaud thank you for the review, have made the change in temp_commit, and added rerun

**Comment by flo-renaud** [comment]
For future ref, link to successful run: [Details](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/e3668418-8606-11ef-9b8d-fa163e929191)

I launched the test on a system without the patch for https://pagure.io/freeipa/issue/9640 and the test correctly detects the issue (see http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/e15fa9e6-87ce-11ef-bc5a-fa163e8ed6f9/report.html?sort=result).

You can remove the temp commit, I will ACK when PRCI completes. Please add "Related: https://pagure.io/freeipa/issue/9640" to your commit message and a small description, for instance:
```
ipatests: Test for ipa hbac rule duplication

Check that ipa-migrate is not creating duplicate default hbac rules
for allow_all and allow_systemd-user rules.

Related: https://pagure.io/freeipa/issue/9640
```

**Comment by menonsudhir** [comment]
> https://pagure.io/freeipa/issue/9640

@flo-renaud thank you for the review, have removed the temp commit and added the description to the commit.

**Comment by rcritten** [comment]
master:

* e1f96ffc2070ba7036c1108c6621450ab8f3f1f5 ipatests: Test for ipa hbac rule duplication

**Pull request closed without merging**